[lte][agw] Updating ipv6 and ipv4v6 handlers on MobilityClient through ITTI msgs

ardzoht opened the pull request:
Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

<!--
    Tag your PR title with the components that it touches.
    E.g. "[lte][agw] Changeset" or "[orc8r][docker] ..."
-->

## Summary

- This PR completes the movement of processing of SPGW-mobilityd IP allocation through ITTI messaging, similarly as was done on https://github.com/magma/magma/pull/5416/ 

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->
- Tested sanity with make integ_test
- Running `test_ipv6_secondary_pdn_rs_retransmit.py` and `test_ipv4v6_secondary_pdn.py` as well as other ipv6 related s1ap integ tests, however tests are failing due to an unrelated issue 

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

ardzoht committed:
Updating ipv6 and ipv4v6 handlers on MobilityClient to send ITTI IP allocation message

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

ardzoht committed:
Updating ipv6 and ipv4v6 handlers on MobilityClient to send ITTI IP allocation message

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

ardzoht commented:
MAGMA-OAI-TESTS

uri200 committed:
[feg] s8_proxy use PdnAddressAllocation instead of string for CreateSessionResponse And adition of GtpError to Responses (#5452)

electronjoe committed:
[AGW] Remove -Werror compiler flag everywhere, add -Wextra (#5358)

Per Discussion [0], here we drop the Werror compiler flag and instead opt for warning monitoring of counts (and possibly increased warning types).

[0] https://github.com/magma/magma/discussions/5273

Note that this lets us turn up warnings, and re-enable some warnings that had been dropped.

An alternative would be to specialize CMake behavior on compiler target - though the existing CMake complexity already causes many problems and I hesitate to introduce further complexity here.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

pshelar committed:
AGW: pipelined: skip old tests. (#5493)

This is causing issues on ubuntu, skip for now.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

pshelar committed:
AGW: pipelined: fix vsctl warning: target type "testing_br" is possibly erroneous (#5491)

ovs-vsctl set-namager does not need bridge name.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

pshelar committed:
AGW: pipelined: uplink-br0: avoid resetting SGi ip address (#5495)

On service restart check if SGi port is part of uplink-br0
before resetting interface IP address.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

pshelar committed:
AGW: Fix memory size for focal VM. (#5492)

I have seen issue while compiling AGW binaries with 6GB so bump it
to 8GB.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

pshelar committed:
AGW: fix pyroute2 dependency. (#5459)

Add specific version.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

ranj8work committed:
[feg]Changes to push FEG docker images (gateway_go|gateway_python) to new JFROG repo (#5466)

* updated to push feg docker images to jfrog repo

Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

* changes to only test feg-build

Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

* fixed typo

Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

* Changes to push feg docker images to new jfrog repo

Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

electronjoe committed:
[CI] Enable GCC Warning Annotations on GH Pull Requests (#5404)

Here is an example GCC Warning Annotation directly on a Pull Request.

<img width="713" alt="Screen Shot 2021-03-14 at 7 24 26 AM" src="https://user-images.githubusercontent.com/3752618/111066627-60449400-8496-11eb-8ac0-be21ec105a72.png">

Among several motivations for this change - I feel that by enabling GCC warning annotations we can now move forward with removal of `-Werror` compiler flags (#5358).

The approach of this PR can now be applied to also annotate Clang 11 warnings, Clang-Tidy findings, and Clang-Format findings (with some extra work required for each - but they will be much easier to add now).

This PR achieves the annotations through the following.

- Dockerfile capable of build and analysis of Magma c/c++ targets
  - Using Ubuntu 20.04
  - GCC 9.3 (default 20.04)
- A Github Workflow that emits Pull Request GCC Warning Annotations
  - That show up directly in-line during code review
  - Using [electronjoe/gcc-problem-matcher](https://github.com/electronjoe/gcc-problem-matcher)

There are annotation limits in GitHub Actions:

- 10 warning annotations and 10 error annotations per step
- 50 annotations per job (sum of annotations from all the steps)
- 50 annotations per run (separate from the job annotations, these annotations aren’t created by users)

Given that our codebase now has 600 GCC 9.3 warnings (with `-Wextra` enabled), this PR had to filter them.  Annotations are filtered and emitted only for modified files.

** The implication is that only the first 10 Annotations will be emitted, for the modified files. **

This could at times miss things, until we burn down some of the GCC warnings being emitted.

In order to determine the set of files modified by the PR - we use the `get-changed-files@v2` github action.  The original implementation (jiterrbit/get-changed-files) has some weaknesses that result in action-terminating warnings under the Magma workflow (any time the forked-branch is not based upon origin-master).  Pragmatically, the warning emitted by jiterrbit's github action does not often impact the output results, so we use here a fork of get-changed-files by mmagician which disables the error / halt under this warning.

** The implication however is that the set of files filtered as having changes in a Pull Request could, in certain situations, not reflect reality due to the complexity of the merge situation. **

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[CMake] Ensure compile_commands.json generation (#5489)

Summary

Because the CMake commands were not being applied within CMakeLists, here we move the configuration of compile_commands.json output to the post-Cmake CMakeCache.txt cofiguration phase.

Test Plan

Executed test build in Docker, observed compile_commands.json in `/build/c/*` for all sub-builds.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

ssanadhya committed:
[AGW][S1ap tests] Split mandatory S1ap tests into precommit and extended (#4798)

* Split mandatory S1ap tests into precommit and extended
* Adding delay in partial eNB test case

Signed-off-by: Shruti Sanadhya <ssanadhya@fb.com>

uri200 committed:
[doc] proposal 008 - Inbound Roaming (#5484)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

themarwhal committed:
[SessionD] Clean up SessionD unit tests by looking at ASAN outputs (#5482)

Signed-off-by: Marie Bremner <marwhal@fb.com>

uri200 committed:
[magmad] Add GRPC logging capabilities to magmad (#5376)

mattymo committed:
Add bare-metal-ansible method for on-prem deploy (#5361)

Signed-off-by: Matthew Mosesohn <matthew.mosesohn@gmail.com>

electronjoe committed:
[Non-Functional] Remove accidentally committed GCC warnings (#5502)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

jbryce committed:
Fixing invalid filename in id attribute (#5513)

Signed-off-by: Jonathan Bryce <jbryce@jbryce.com>

uri200 committed:
[orc8r] Make s8 field optional in swagger for FEG gateways too (#5512)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

drbruced12 committed:
Update node version for docusaurus (#5511)

Signed-off-by: Bruce Davie <3101026+drbruced12@users.noreply.github.com>

ranj8work committed:
Changes to push cwf and orc8r docker images to the new JFROG repo (#5504)

Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

tmdzk committed:
Increase fab timeout for lte_integ_test (#5523)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

tmdzk committed:
Split slack notifications (#5522)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

AyliD committed:
[xwf][radius] take AAA_ENDPOINT from enviornment (#5528)

Signed-off-by: Ayelet Regev Dabah <ayelet.regev@gmail.com>

tmdzk committed:
[AGW][Baremetal Install]Master to point to stretch-1.4.0 (#5519)

* Master to point to stretch-1.4.0

Signed-off-by: Timothée Dzik <timdzik@fb.com>

* Master to point to stretch-1.4.0

Signed-off-by: Timothée Dzik <timdzik@fb.com>

themarwhal committed:
Save commit hash and add to env var (#5530)

Signed-off-by: Marie Bremner <marwhal@fb.com>

themarwhal committed:
[SessionD] Modify SessionD unit tests for better stability (#5534)

Signed-off-by: Marie Bremner <marwhal@fb.com>

pshelar committed:
AGW: qos: fix ovs meter implementation. (#5533)

During TC-QoS enhancements, some of meter api were broken.
Following patch fixes it

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

ssanadhya committed:
Remove unused S1AP ITTI log msgs and gcc warning on fn return type (#5536)

Signed-off-by: Shruti Sanadhya <ssanadhya@fb.com>

koolzz committed:
[pipelined]Temporarily disable CWF qos (#5537)

* Temporarily skip CWF qos

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

* Update test yml

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

Commit:
Cut version v1.4 docs (#5474)

Signed-off-by: Michael Germano <mgermano@fb.com>

rdefosse committed:
[MME][Ubuntu18] fixes for build oai-mme in RelWithDebInfo (#4766)

* [MME][Ubuntu18] fixes for build oai-mme in RelWithDebInfo more

Signed-off-by: Raphael Defosseux <raphael.defosseux@openairinterface.org>

* [MME][OAI] changes after peer review

Signed-off-by: Raphael Defosseux <raphael.defosseux@openairinterface.org>

ssanadhya commented on lte/gateway/c/oai/lib/mobility_client/MobilityClientAPI.cpp:
```diff
@@ -197,101 +191,84 @@ int pgw_handle_allocate_ipv6_address(
-        }
-        OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNok);
+        handle_allocate_ipv6_address_status(
+            status, ip6_addr, vlan, subscriber_id, apn, pdn_type, context_teid,
```
As with Ipv4 changes, please pass a String object instead of pointer for subscriber_id, apn and pdn_type

ssanadhya commented on lte/gateway/c/oai/lib/mobility_client/MobilityClientAPI.cpp:
```diff
@@ -306,83 +283,59 @@ int pgw_handle_allocate_ipv4v6_address(
+        memcpy(&ip6_addr, ipv6_addr_str.c_str(), sizeof(in6_addr));
+        int vlan = atoi(ip_msg.vlan().c_str());
+        handle_allocate_ipv4v6_address_status(
+            status, ip4_addr, ip6_addr, vlan, subscriber_id, apn, pdn_type,
```
Same comment as above

ssanadhya commented on lte/gateway/c/oai/lib/mobility_client/MobilityClientAPI.cpp:
```diff
@@ -306,83 +283,59 @@ int pgw_handle_allocate_ipv4v6_address(
+            context_teid, eps_bearer_id);
       });
-  OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNok);
+  return 0;
```
```suggestion
  return RETURNok;
```

koolzz committed:
[pipelined]Remove UE controller check to unblock CI (#5541)

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

drbruced12 committed:
correct links to images in feg setup guide (#5518)

Signed-off-by: Bruce Davie <3101026+drbruced12@users.noreply.github.com>

AyliD committed:
[xwf][radius] take AAA_ENDPOINT from enviornment to compose (#5558)

Signed-off-by: Ayelet Regev Dabah <ayelet.regev@gmail.com>

tmdzk committed:
COmment out new registry push due to unexpected 500 (#5563)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

electronjoe committed:
Remove -Wextra from default flags, enable env use (#5559)

This PR removes -Wextra from default flags, in order to mitigate the high warning count currently being emitted in our builds.

It is added to Github Action Annotations, however, by use of env parameters to the Makefile. This enables Github pull requests to still benefit from -Wextra.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

themarwhal committed:
[SessionD] Remove boolean return values from PipelinedClient (#5472)

Signed-off-by: Marie Bremner <marwhal@fb.com>

uri200 committed:
[mme] Add federated map to mme.config and template (#5552)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

ShubhamTatvamasi committed:
[orc8r][helm] Fixed spelling mistake on keywords (#5574)

Signed-off-by: Shubham Tatvamasi <shubhamtatvamasi@gmail.com>

uri200 committed:
[feg] enable S8_proxy on docker-compose (#5058)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

uri200 committed:
[mme] fix generate_oai_config.py (#5585)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

electronjoe committed:
[session_manager] Add missing headers and linking for GPR_ASSERT (#5562)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

GPR_ASSERT technically requires some header includes that were missing. Further, in my new docker developer environment, we failed linking for this target due to lack of link with -lgpr.

Working goes toward #5529.

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
Fixup missing includes detected by Clang (#5560)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

These build targets failed in my (likely newer) docker environment with GCC 9.3 / CMake latest of Ubuntu 20.04. We should IncludeWhatYouUse this whole directory. Here I just spot fix to get compiling myself.

Working goes toward #5529.

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[Coverage] Add BUILD_TESTS flags for code coverage output (#5561)

* [session_manager] Add flags to enable code coverage for tests

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

* [connection_tracker] Add flags to enable code coverage for tests

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

prabinakpattnaik committed:
Add Pipelined Classifier IPv6 support #4556 (#5275)

* Add Pipelined Classifier IPv6 support #4556

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* Add Pipelined Classifier IPv6 support #4556

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* Add Pipelined Classifier IPv6 support #4556

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* Add Pipelined Classifier IPv6 support #4556

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* Add Pipelined Classifier IPv6 support #4556

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

AndyLKhuu committed:
[orc8r] [fix] Register a Swagger spec servicer for testcontroller (#5579)

Signed-off-by: Andy Lee Khuu <andykhuu@stanford.edu>

AndyLKhuu committed:
[orc8r] Add simple collapse button (#5538)

Signed-off-by: Andy Lee Khuu <andykhuu@stanford.edu>

ymasmoudi committed:
[xwf-m] fix trailing \t error in docker compose (#5588)

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

ymasmoudi committed:
[xwf-m] validate yaml prior to upgrade (#5589)

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

hcgatewood committed:
[orc8r][docs] Move state indexer docs to Docusaurus (#5430)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

rsarwad committed:
Add s8 contexts in AGW (#5270)

* Added S8 contexts and framework for stateless S8 task

Signed-off-by: rashmi <rashmi.sarwad@radisys.com>

Vyas20 committed:
[Converged-MME]mobile_reachability_timer and implicit_detach_timer edit. (#4802)

* replacing mobile_reachability_timer and implicit_detach_timer

Signed-off-by: Mohit Vyas <mohit.vyas@openairinterface.org>

Co-authored-by: lionelgo <29477918+lionelgo@users.noreply.github.com>

AndyLKhuu committed:
[agw] Attach Optional Sentry Error Handling (#5566)

AndyLKhuu committed:
[docs] [orc8r] Add docs for module-module dependencies (#5503)

Signed-off-by: Andy Lee Khuu <andykhuu@stanford.edu>

hcgatewood committed:
[scripts] Add script to pull Magma codeowner info (#5578)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

hcgatewood committed:
[orc8r][docs] Add new contributor docs (#5384)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

themarwhal committed:
[SessionD] Activate all rules with PolicyRule definitions (#5575)

pshelar committed:
AGW: VM: virtual box clock fix (#5619)

electronjoe committed:
[MME] Run Clang-Tidy on master, push results to Spreadsheet (#5354)

This PR executes Clang-Tidy analysis on merges to master.  Detailed logs are available, but key fidnings are also tabulated and pushed to the following Public Google Spreadsheet for tracking over time.

[MME Clang-Tidy Historical Data](https://docs.google.com/spreadsheets/d/e/2PACX-1vRYjUoXzntoIQy-MH_0rI97yO4EQ0h-FFM67AqzIztax0cbnatRBmOs9teKOPcZUQ06qUAY9RN4qQfG/pubhtml)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

Vyas20 committed:
Updating the chronology of exiting the tasks and czmq (#5623)

Signed-off-by: Mohit Vyas <mohit.vyas@openairinterface.org>

talkhasib committed:
[MME][Bug] On MME shutdown, stop GRPC service before releasing ZMQ context (#5542)

Signed-off-by: Tariq Al-Khasib <talkhasib@fb.com>

arunuke committed:
Building artifacts for Cloudstrapper (#5618)

Signed-off-by: Arun Thulasi <arunt@fb.com>

tmdzk committed:
[AGW] Install script using shell in bash (#5545)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

tmdzk committed:
Only run doc on doc changes (#5610)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

electronjoe committed:
[CI] Add a suite of useful GCC Warning flags (#5591)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[CI] Enable c/c++ codecov support (#5529)

Using the Docker container built for GCC Annotations in GH (#5404), build and run the following tests, then compute code coverage and push to codecov.io.

- test_oai
- test_session_manager

Test Plan

Submitted PR and observed [codecov.io reported results](https://codecov.io/gh/magma/magma/tree/45b981c738c07adf83bf474430358f0d1c69f51b/lte/gateway/c).

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

Vyas20 committed:
[Converged MME]NSA-Addition (#4513)

* Adding messages and IEs for NSA SUPPORT.
* .proto flags deprecated

Signed-off-by: Mohit Vyas <mohit.vyas@openairinterface.org>

ulaskozat committed:
Revert ICS message criticality to reject (#5626)

Signed-off-by: Ulas Kozat <kozat@fb.com>

uri200 committed:
[feg] Add code coverage for feg and addto CircleCi (#5630)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

karthiksubraveti committed:
Swagger changes for Restricted IMEIs (#5631)

Signed-off-by: Karthik Subraveti <ksubraveti@fb.com>

themarwhal committed:
[MME] Add a basic spgw-task state_conversion test (#5509)

ssanadhya committed:
Read back ULR response timer from Redis (#5637)

Vyas20 committed:
[Converged-MME]ulr_response_timer edit. (#4807)

* ulr_response_timer edit

Signed-off-by: Mohit Vyas <mohit.vyas@openairinterface.org>

Vyas20 committed:
[Converged-MME]ue_context_modification_timer edit. (#4803)

* replacing ue_context_modification_timer

Signed-off-by: Mohit Vyas <mohit.vyas@openairinterface.org>

themarwhal committed:
[AGW][common][sctpd] Add back unit tests in sctpd and magma_common and move some OAI tests to magma_common (#5414)

* Add make commands to run magma_common unit tests

Signed-off-by: Marie Bremner <marwhal@fb.com>

* Add SCTPD tests to make test

Signed-off-by: Marie Bremner <marwhal@fb.com>

* Move service registry / metrics / service303 unit tests from OAI to magma_common

Signed-off-by: Marie Bremner <marwhal@fb.com>

* Address clang-tidy warnings in common C++ directory

Signed-off-by: Marie Bremner <marwhal@fb.com>

electronjoe committed:
[CI] only run c/c++ build and test when modified (#5629)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

hcgatewood committed:
[codeowners] Add Karthik, Hanna, Andrei (#5636)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

electronjoe committed:
Update gitignore for build directories (#5218)

This change would limit gitignore to only `/magma/build`.

I am unsure whether build directories are ever generated anywhere but `/magma/build`.

I do this because the present gitignore causes shadowing of e.g. `third_party/build/...` subdirectory - which clearly should not be gitignored.

If the are build directories generated elsewhere, then we need to rename `third_party/build`.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[MME] fix lib/hashtable undersized calloc of key structure (#5327)

This is two of ~ten Clang-Tidy findings for check type clang-analyzer-unix.MallocSizeof.

- hashtable.c:460
  - Results in under-allocation of hashtable keys if sizeof(hash_key_t*) != sizeof(uint64_t)
    - E.g. 32 bit builds
- hashtable.c:492
  - Results in under-allocation of hashtable keys if sizeof(hash_key_t*) !+ sizeof(void*)
    - Not guaranteed by [C++11 draft standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf) Section 6.2.5 Types para 28

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[MME] fix hashtable/hashtable_uint64.c potentially undersized calloc of key structure (#5328)

This is two of ~ten Clang-Tidy findings for check type clang-analyzer-unix.MallocSizeof (see #5326).

- hashtable_uint64.c:436
  - Results in under-allocation of hashtable keys if sizeof(hash_key_t*) != sizeof(hash_key_t)
    - E.g. 32 bit builds
- hashtable_uint64.c:466
  - Results in under-allocation of hashtable keys if sizeof(uint64_t*) != sizeof(uint64_t)
    - E.g. 32 bit builds

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[MME] Correct mis-sized allocation of grpc_service_data_t (#5329)

Incorrect use of sizeof() reference here was probably guarded by the fact that grpc_service_data_t is strictly larger than the intended grpc_service_config struct.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[MME] Fix null pointer dereference in TLVDecoder.c (#5335)

Summary

In this code branch, `bstr == NULL`.  Removing the assignment of `*bstr = NULL.`

Test Plan

```
cd lte/gateway
make build_oai
```

And CI Pipelines.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[docs] Update guidance for Docker memory allocation (#4564)

The Docker UI has evolved, update the instructions and screenshot.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

electronjoe committed:
[MME] IWYU applied to test/openflow subdir (#5226)

A first example of the output from IncludeWhatYouUse (Issue #4868) applied to the test/openflow subdirectory.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

koolzz committed:
Remove Static Rule Processing from Pipelined(#5624)

We've moved to only dynamic rule approach where sessiond automatically looks ups static rules for us. This PR removes all processing of the static rules on pipelined side.


Commit log:

* [pipelined]Remove static rule processing from

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

* Fix lint

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

* Update feg test

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

* Fix bug

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

ardzoht committed:
Updates based on comments

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

ardzoht committed:
Merge branch 'add_ipv6_ip_allocation_itti_msg' of github.com:ardzoht/magma into add_ipv6_ip_allocation_itti_msg

* 'add_ipv6_ip_allocation_itti_msg' of github.com:ardzoht/magma:
  Updating ipv6 and ipv4v6 handlers on MobilityClient to send ITTI IP allocation message
